### ISSUE TITLE
Debian 9 - udev error on vmware scsi udev rule

### ISSUE THREAD

**Issue by artn** [vmware/open-vm-tools#183]
Hi, I installed debian 9 and found below log entry in /var/log/damen.log.

```
Jul  6 13:32:18 localhost systemd-udevd[290]: Process '/bin/sh -c 'echo 180 >/sys$DEVPATH/device/timeout'' failed with exit code 2.
```

It seems that applying timeout setting for device was failed. So I'd like to know is there any side effect if I use debian 9 as it is.

**Comment by shaba** [comment]
Hello. I was fix this error in my guest with udev rules:
cat /lib/udev/rules.d/99-vmware-scsi-udev.rules
ACTION=="add", SUBSYSTEMS=="scsi", ENV{DEVTYPE}=="scsi_device", ATTRS{vendor}=="VMware*" , ATTRS{model}=="Virtual disk*", ATTR{timeout}="180"
ACTION=="add", SUBSYSTEMS=="scsi", ENV{DEVTYPE}=="scsi_device", ATTRS{vendor}=="VMware*" , ATTRS{model}=="VMware Virtual S", ATTR{timeout}="180"

**Comment by artn** [comment]
@shaba 
Thank you for comment. Since there is no description about `ENV{DEVTYPE}=="scsi_device"` in [original file](https://github.com/vmware/open-vm-tools/blob/master/open-vm-tools/udev/99-vmware-scsi-udev.rules) so I think it's better to remove it and successfully set timeout parameter by below script.

```
ACTION=="add", SUBSYSTEMS=="scsi", ATTRS{vendor}=="VMware*" , ATTRS{model}=="Virtual disk*", ATTR{timeout}="180"
ACTION=="add", SUBSYSTEMS=="scsi", ATTRS{vendor}=="VMware*" , ATTRS{model}=="VMware Virtual S", ATTR{timeout}="180"
```

**Comment by shaba** [comment]
without ENV{DEVTYPE}=="scsi_device", i had error:
`error opening ATTR{/sys/devices/pci0000:00/0000:00:15.0/0000:03:00.0/host0/target0:0:0/0:0:0:0/bsg/0:0:0:0/timeout} for writing: Permission denied
error opening ATTR{/sys/devices/pci0000:00/0000:00:15.0/0000:03:00.0/host0/target0:0:0/0:0:0:0/scsi_device/0:0:0:0/timeout} for writing: Permission den
ied
error opening ATTR{/sys/devices/pci0000:00/0000:00:15.0/0000:03:00.0/host0/target0:0:0/0:0:0:0/scsi_disk/0:0:0:0/timeout} for writing: Permission denie
d
error opening ATTR{/sys/devices/pci0000:00/0000:00:15.0/0000:03:00.0/host0/target0:0:1/0:0:1:0/bsg/0:0:1:0/timeout} for writing: Permission denied
error opening ATTR{/sys/devices/pci0000:00/0000:00:15.0/0000:03:00.0/host0/target0:0:1/0:0:1:0/scsi_device/0:0:1:0/timeout} for writing: Permission den
ied
error opening ATTR{/sys/devices/pci0000:00/0000:00:15.0/0000:03:00.0/host0/target0:0:1/0:0:1:0/scsi_disk/0:0:1:0/timeout} for writing: Permission denie
d
error opening ATTR{/sys/devices/pci0000:00/0000:00:15.0/0000:03:00.0/host0/target0:0:0/0:0:0:0/block/sda/timeout} for writing: Permission denied
error opening ATTR{/sys/devices/pci0000:00/0000:00:15.0/0000:03:00.0/host0/target0:0:0/0:0:0:0/block/sda/sda2/timeout} for writing: Permission denied
error opening ATTR{/sys/devices/pci0000:00/0000:00:15.0/0000:03:00.0/host0/target0:0:0/0:0:0:0/block/sda/sda1/timeout} for writing: Permission denied
error opening ATTR{/sys/devices/pci0000:00/0000:00:15.0/0000:03:00.0/host0/target0:0:1/0:0:1:0/block/sdb/timeout} for writing: Permission denied
error opening ATTR{/sys/devices/pci0000:00/0000:00:15.0/0000:03:00.0/host0/target0:0:1/0:0:1:0/block/sdb/sdb1/timeout} for writing: Permission denied
`

**Comment by artn** [comment]
@shaba 
You are right.
I also found error log entries in /var/log/syslog and Fixed by adding `ENV{DEVTYPE}=="scsi_device"`, .

**Comment by bzed** [comment]
@artn @shaba actually - as far as I can see - all scsi disks have proper timeout settings. I'm not exactly sure about which devices udev complains here, but your fix in the pull request of the debian package definitely makes sense. You might want to send that pull request toe this upstream repository, too!

**Comment by oliverkurth** [comment]
We have a fix for this in the devel branch with commit 5c77f47.

**Comment by oliverkurth** [comment]
Oh, I am sorry. After almost a year I have no idea why I sent that.

**Comment by takenek** [comment]
But problem still exist in Debian 9:

root@zabbix:~# dpkg -l |grep open-vm-tools
ii  open-vm-tools                    2:10.1.5-5055683-4+deb9u1      amd64        Open VMware Tools for virtual machines hosted on VMware (CLI)

Oct 19 23:19:27 zabbix systemd[1]: Started Monitoring of LVM2 mirrors, snapshots etc. using dmeventd or progress polling.
Oct 19 23:19:27 zabbix systemd-udevd[349]: Process '/bin/sh -c 'echo 180 >/sys$DEVPATH/device/timeout'' failed with exit code 2.
Oct 19 23:19:27 zabbix systemd-udevd[340]: Process '/bin/sh -c 'echo 180 >/sys$DEVPATH/device/timeout'' failed with exit code 2.
Oct 19 23:19:27 zabbix systemd[1]: Found device VMXNET3 Ethernet Controller.
Oct 19 23:19:27 zabbix systemd-udevd[341]: Process '/bin/sh -c 'echo 180 >/sys$DEVPATH/device/timeout'' failed with exit code 2.
Oct 19 23:19:27 zabbix systemd[1]: Started Set the console keyboard layout.
Oct 19 23:19:27 zabbix systemd[1]: Reached target Local File Systems (Pre).
Oct 19 23:19:27 zabbix systemd-udevd[342]: Process '/bin/sh -c 'echo 180 >/sys$DEVPATH/device/timeout'' failed with exit code 2.
Oct 19 23:19:27 zabbix systemd-udevd[349]: Process '/bin/sh -c 'echo 180 >/sys$DEVPATH/device/timeout'' failed with exit code 2.
Oct 19 23:19:27 zabbix systemd[1]: Found device Virtual_disk 1.
Oct 19 23:19:27 zabbix systemd[1]: Starting File System Check on /dev/disk/by-uuid/bd73bff6-cd44-43a7-9727-e9e0dde29dc9...
Oct 19 23:19:27 zabbix systemd-udevd[336]: Process '/bin/sh -c 'echo 180 >/sys$DEVPATH/device/timeout'' failed with exit code 2.

Best Regards

**Comment by bjmi** [comment]
Same with Ubuntu 18.04
```
root@ubuntusrv:~# dpkg -l |grep open-vm-tools
ii  open-vm-tools                         2:10.3.0-0ubuntu1~18.04.3                  amd64        Open VMware Tools for virtual machines hosted on VMware (CLI)
```
```
Mar 16 12:09:30 ubuntusrv systemd-udevd[12373]: Process '/bin/sh -c 'echo 180 >/sys$DEVPATH/timeout'' failed with exit code 2.
Mar 16 12:09:30 ubuntusrv systemd-udevd[12369]: Process '/bin/sh -c 'echo 180 >/sys$DEVPATH/timeout'' failed with exit code 2.
Mar 16 12:09:30 ubuntusrv systemd-udevd[12355]: Process '/bin/sh -c 'echo 180 >/sys$DEVPATH/timeout'' failed with exit code 2.
Mar 16 12:09:30 ubuntusrv systemd-udevd[12358]: Process '/bin/sh -c 'echo 180 >/sys$DEVPATH/timeout'' failed with exit code 2.
Mar 16 12:09:30 ubuntusrv systemd-udevd[12371]: Process '/bin/sh -c 'echo 180 >/sys$DEVPATH/timeout'' failed with exit code 2.
Mar 16 12:09:30 ubuntusrv systemd-udevd[12370]: Process '/bin/sh -c 'echo 180 >/sys$DEVPATH/timeout'' failed with exit code 2.
Mar 16 12:09:30 ubuntusrv systemd-udevd[12387]: Process '/bin/sh -c 'echo 180 >/sys$DEVPATH/timeout'' failed with exit code 2.
Mar 16 12:09:30 ubuntusrv systemd-udevd[12415]: Process '/bin/sh -c 'echo 180 >/sys$DEVPATH/timeout'' failed with exit code 2.
```

**Comment by stanguturi** [comment]
> and will try https://bugs.launchpad.net/ubuntu/+source/open-vm-tools/+bug/1668244/+attachment/5088336/+files/1668244_udev_rule_60-open-vm-tools.rules.patch
> from https://bugs.launchpad.net/ubuntu/+source/open-vm-tools/+bug/1668244

This is fixed in commit 3011466e0f59471292a58ba5124238898d18945a Closing this issue.

Please check https://github.com/vmware/open-vm-tools/commits/master/open-vm-tools/udev/99-vmware-scsi-udev.rules for the entire commit history of udev rule file.

Thanks
Sankar.
